### PR TITLE
[Content] clarify Image to labels, Labels to Image, and add docs for shapes to labels behavior

### PR DESCRIPTION
**Pull request by psobolewskiPhD**
# References and relevant issues
Depends on napari/napari#6859

# Description
This PR clarifies the Image to Labels and Labels to Image contextual menu actions, plus adds documentation for the altered Shapes to Labels behavior introduced by https://github.com/napari/napari/pull/6859

### CODE REVIEW

File: docs/tutorials/fundamentals/viewer.md [hunk 1]
```diff
@@ -478,8 +478,8 @@ The right side of the status bar contains some helpful tips depending on which l
 * **Show All Unselected Layers** - Set all *unselected* layers to visible.
 * **Hide All Unselected Layers** - Set all *unselected* layers to hidden.
 * **Duplicate Layer** - creates a second copy of the selected layer. Can be used on **Points**, **Shapes**, **Labels**, and **Image** layers. This is useful for testing your analysis on a copy instead of on the original image.
-* **Convert to Labels** - converts an **Image** layer to a **Labels** layer. This is useful for converting a binary image segmentation map to a labels layer with each segmented object denoted by its own integer. Can also be used on a **Shapes** layer.
-* **Convert to Image** - converts a **Labels** layer into an **Image** layer.
+* **Convert to Labels** - converts an **Image** layer to a **Labels** layer (the data is not copied or altered, only the representation is changed, making this fully reversible). This is useful for converting a binary image segmentation map to a labels layer with each segmented object denoted by its own integer. Can also be used on a **Shapes** layer, in this case a new **Labels** layer will be created.
```
Comment (Czaki) [inline docs/tutorials/fundamentals/viewer.md]: I just spot, that it is not fully reversible for float images.

**Review by Czaki — APPROVED** [review]

File: docs/tutorials/fundamentals/viewer.md (hunk 1 above)
Comment (psobolewskiPhD) [reply docs/tutorials/fundamentals/viewer.md]: oh yeah float would be an issue right, because labels are int.
annoying

Comment (psobolewskiPhD) [reply docs/tutorials/fundamentals/viewer.md]: I pushed a change to say only if image is int is it not copied--to match my push to the napari PR. What do you think?

Comment (Czaki) [reply docs/tutorials/fundamentals/viewer.md]: looks ok for me

**Comment by psobolewskiPhD** [comment]
testing comment

**Comment by TimMonko** [comment]
Thanks for this! Had some unexpected behavior today, with the image-to-labels layer being added instead of popped. Turned out its the difference between float and int image dtypes, and this doc change was very helpful!

**Pull request merged**